server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -324,6 +325,38 @@ public void testEmptyDatasource()
     );
   }
 
+
+  @Test
+  public void testWhitespaceDatasource()
+  {
+    String[] dataSource = {"\tab\t", "\rcarriage\return\r", "\nnew\nline\n"};
```

Comment:
Consider using a `Map` instead to keep the respective `dataSource` and `invalidChars` elements paired correctly. For example:
```java
Map<String, String> invalidCharToDataSourceName = ImmutableMap.of(
    "\\t", "\tab\t",
    "\\r", "\rcarriage\return\r",
    "\\n", "\nnew\nline\n"
);

// iterate over map entries
```